ls1intum/Artemis
# `Development`: Move docker image pulls for programming exercises from startup to first build for local continuous integration

### Checklist
#### General


#### Server

### Motivation and Context


The pulling of docker images should not happen during the server startup as it delays the whole process. Docker images should be pulled when they are actually needed.

### Description
This PR essentially moves some image pull logic from the Local CI configuration (which is initialized on startup) to the build job management service.

Before a build job is added to the build job queue, Artemis checks whether the image already exists. If not, it is pulled. Thus, the image is pulled the first time a programming exercise using that template is created.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Make sure the image `ls1tum/artemis-maven-template:java17-18` is removed from your local machine
2. Start the LocalVC & LocalCI setup on your local machine
3. Check whether the image is pulled during server startup (it shouldn't be)
4. Create a programming exercise
5. Check whether is pulled during exercise creation


### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
